### PR TITLE
bazci: fix staging behavior

### LINKED ISSUE #86773: bazci: can collect .xml files from prior benchmark runs
Example: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/6208678?buildTab=overview&showRootCauses=false&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true#%2Fbazel-testlogs%2Fpkg%2Fccl%2Fchangefeedccl%2Fchangefeedccl_test

I suspect that this build agent ran a benchmark build configuration before running the unit tests, and the `test.xml` files from that run were leftover in `bazel-testlogs`. Then the unit test run came and picked up on these old `test.xml` files, not realizing they were stale.

Epic CRDB-15060

Jira issue: CRDB-18924

### PR DESCRIPTION
**Pull request by rickystewart**
Up until this point `bazci` has trawled the `bazel-bin` and
`bazel-testlogs` directories to find output artifacts for staging, but
this is not what we should be doing, as in doing so we're likely to find
stale, out-of-date files that belong to old builds or test runs.

We fix this by instead checking the
[build event protocol log](https://bazel.build/remote/bep), which
contains all the information we need for staging test results and built
binaries.

Closes #86773.

Release justification: Non-production code changes
Release note: None

### CODE REVIEW

File: pkg/cmd/bazci/bazci.go [hunk 1]
```diff
@@ -347,30 +139,132 @@ func mergeTestXMLs(args parsedArgs) error {
...
+						dst := filepath.Join(artifactsDir, outputDir, filepath.Base(src))
+						if err := doCopy(src, dst); err != nil {
+							return err
+						}
+					} else {
+						panic(output)
+					}
+				}
 			}
+		default:
```
Comment (healthy-pod) [inline pkg/cmd/bazci/bazci.go:209]: nit: any reason for keeping an empty default?

File: pkg/cmd/bazci/bazci.go [hunk 2]
```diff
@@ -57,262 +68,43 @@ func init() {
...
 		return err
 	}
+	bepLoc := filepath.Join(tmpDir, "beplog")
+	args = append(args, fmt.Sprintf("--build_event_binary_file=%s", bepLoc))
+	fmt.Println("running bazel w/ args: ", shellescape.QuoteCommand(args))
+	bazelCmd := exec.Command("bazel", args...)
+	bazelCmd.Stdout = os.Stdout
+	bazelCmd.Stderr = os.Stderr
+	err = bazelCmd.Run()
+	// ? What to do with error
```
Comment (healthy-pod) [inline pkg/cmd/bazci/bazci.go:98]: Printing it and continuing LGTM but maybe we should add a warning to the error message that the `bep binary file` may not have been created to guide future debugging?

**Review by healthy-pod — APPROVED** [review]

**Comment by rickystewart** [comment]
bors r=healthy-pod

File: pkg/cmd/bazci/bazci.go [hunk 3]
```diff
@@ -347,30 +139,131 @@ func mergeTestXMLs(args parsedArgs) error {
...
+			summary := event.GetTestSummary()
+			for _, testResult := range testResults[label] {
+				if testResult.run > 1 {
+					outputDir = filepath.Join(outputDir, fmt.Sprintf("run_%d", testResult.run))
+				}
+				if summary.ShardCount > 1 {
+					outputDir = filepath.Join(outputDir, fmt.Sprintf("shard_%d_of_%d", testResult.shard, summary.ShardCount))
+				}
+				if testResult.attempt > 1 {
+					outputDir = filepath.Join(outputDir, fmt.Sprintf("attempt_%d", testResult.attempt))
```
Comment (srosenberg) [inline pkg/cmd/bazci/bazci.go:195]: Nice!

**Pull request merged**